From 8b3fcf0231052011f82febda3040149fe48fe147 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 4 Aug 2020 13:19:35 +0800
Subject: [PATCH] policy/modules/system/userdomain: allow domain read home link
 directory

Fixes:
avc:  denied  { read } for  pid=453 comm="alsactl" name="root"
dev="sdb3" ino=1600 scontext=system_u:system_r:alsa_t:s0-s15:c0.c1023
tcontext=root:object_r:user_home_dir_t:s0 tclass=lnk_file permissive=0

avc:  denied  { read } for  pid=1494 comm="lpqd" name="root" dev="sdb3"
ino=1600 scontext=system_u:system_r:smbd_t:s0-s15:c0.c1023
tcontext=root:object_r:user_home_dir_t:s0 tclass=lnk_file permissive=0

Upstream-Status: Inappropriate [ostree specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/userdomain.if | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/policy/modules/system/userdomain.if b/policy/modules/system/userdomain.if
index b04f40a41..f27002dca 100644
--- a/policy/modules/system/userdomain.if
+++ b/policy/modules/system/userdomain.if
@@ -1673,6 +1673,7 @@ interface(`userdom_getattr_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir getattr_dir_perms;
 	files_search_home($1)
 ')
@@ -1710,6 +1711,7 @@ interface(`userdom_search_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -1755,6 +1757,7 @@ interface(`userdom_list_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir list_dir_perms;
 	files_search_home($1)
 ')
@@ -1792,6 +1795,7 @@ interface(`userdom_create_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir create_dir_perms;
 ')
 
@@ -1810,6 +1814,7 @@ interface(`userdom_manage_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir manage_dir_perms;
 ')
 
@@ -1828,6 +1833,7 @@ interface(`userdom_relabelto_user_home_dirs',`
 		type user_home_dir_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir relabelto;
 ')
 
@@ -1890,6 +1896,7 @@ interface(`userdom_user_home_domtrans',`
 	')
 
 	domain_auto_transition_pattern($1, user_home_t, $2)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -2338,6 +2345,7 @@ interface(`userdom_manage_user_home_content_files',`
 	')
 
 	manage_files_pattern($1, user_home_t, user_home_t)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -2378,6 +2386,7 @@ interface(`userdom_manage_user_home_content_symlinks',`
 	')
 
 	manage_lnk_files_pattern($1, user_home_t, user_home_t)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -2437,6 +2446,7 @@ interface(`userdom_manage_user_home_content_pipes',`
 	')
 
 	manage_fifo_files_pattern($1, user_home_t, user_home_t)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -2457,6 +2467,7 @@ interface(`userdom_manage_user_home_content_sockets',`
 		type user_home_dir_t, user_home_t;
 	')
 
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	manage_sock_files_pattern($1, user_home_t, user_home_t)
 	files_search_home($1)
@@ -2494,6 +2505,7 @@ interface(`userdom_user_home_dir_filetrans',`
 		type user_home_dir_t;
 	')
 
+	read_lnk_files_pattern($1, user_home_dir_t, user_home_dir_t)
 	filetrans_pattern($1, user_home_dir_t, $2, $3, $4)
 	files_search_home($1)
 ')
@@ -2532,6 +2544,7 @@ interface(`userdom_user_home_content_filetrans',`
 	')
 
 	filetrans_pattern($1, user_home_t, $2, $3, $4)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 user_home_dir_t:dir search_dir_perms;
 	files_search_home($1)
 ')
@@ -4342,6 +4355,7 @@ interface(`userdom_search_user_home_content',`
 	')
 
 	files_list_home($1)
+	allow $1 user_home_dir_t:lnk_file read_lnk_file_perms;
 	allow $1 { user_home_dir_t user_home_t }:dir search_dir_perms;
 ')
 
-- 
2.17.1

